script/post_split.rb

```diff
@@ -0,0 +1,123 @@
+REPLY_ATTRS = [:character_id, :icon_id, :character_alias_id, :user_id, :content, :created_at, :updated_at].map(&:to_s)
+POST_ATTRS = [:board_id, :section_id, :privacy, :status, :authors_locked].map(&:to_s)
+
+def split_post
+  print('Subject for new post? ')
+  new_subject = STDIN.gets.chomp
+  raise RuntimeError, "Invalid subject" if new_subject.blank?
+
+  print("\n")
+  print('First reply_id of new post? ')
+  reply_id = STDIN.gets.chomp
+
+  Post.transaction do
+    first_reply = Reply.find_by(id: reply_id)
+    raise RuntimeError, "Couldn't find reply" unless first_reply
+    old_post = first_reply.post
+    puts "splitting post #{old_post.id}: #{old_post.subject}, at #{reply_id}"
+
+    other_replies = old_post.replies.where('reply_order > ?', first_reply.reply_order).ordered
+    puts "ie starting at + onwards from #{first_reply.inspect}"
+    new_post = create_post(first_reply, old_post: old_post, subject: new_subject)
+
+    new_authors = find_authors(other_replies)
+    migrate_replies(other_replies, new_post: new_post, old_post: old_post, first_reply: first_reply)
+    cleanup_first(first_reply)
+    update_authors(new_authors, new_post: new_post, old_post: old_post)
+    update_caches(new_post, new_post.replies.ordered.last)
+    update_caches(old_post, old_post.replies.ordered.last)
+  end
+end
+
+def create_post(first_reply, old_post:, subject:)
+  new_post = Post.new(first_reply.attributes.slice(*REPLY_ATTRS))
+  new_post.skip_edited = true
+  new_post.is_import = true
+  new_post.assign_attributes(old_post.attributes.slice(*POST_ATTRS))
+  new_post.subject = subject
+  new_post.edited_at = first_reply.updated_at
+  puts "new post: #{new_post.inspect}"
+  new_post.save!
+  puts "new post: https://glowfic.com/posts/#{new_post.id}"
+  new_post
+end
+
+def find_authors(other_replies)
+  author_ids = other_replies.except(:order).select(:user_id).distinct.pluck(:user_id)
+  author_ids.to_h { |id| [id, other_replies.find_by(user_id: id)] }
```

Comment:
We're just using this for the `.created_at` value, right? I might return that in the hash directly + use that below, to make this easier to follow (I was wondering why we'd select some random reply for the author 😅 but it turns out this is ordered and we only use the created_at)

Reply:
This was in fact your fault, I just refactored it xD

Reply:
I'm super good at this whole "writing bogus code and coming back to it and realizing it's bogus" thing :D my bad